### PR TITLE
Adds Debug option for hpx initializing from main

### PR DESCRIPTION
**Pull request by NK-Nikunj**
## Proposed Changes

This PR adds Debug option for `hpx` initialization after using `hpx_main.hpp`. Consider the following example:

```cpp
#include <hpx/hpx_init.hpp>
#include <hpx/hpx_main.hpp>

int hpx_main()
{
  return hpx::finalize();
}

int main()
{
  return hpx::init();
}
```

Since `hpx` runtime is already initialized from main, this would result in the following error message:

```
hpx is already initialized from main.
Note: Delete hpx_main.hpp to initialize hpx system using hpx::init. Exiting...
```

This is more verbose as compared to previous error message which would explain that `hpx::init can't initialize the runtime system.`

### CODE REVIEW

File: src/hpx_init.cpp [hunk 1]
```diff
@@ -584,6 +595,19 @@ namespace hpx
 
             int result = 0;
             try {
+                #if (HPX_HAVE_DYNAMIC_HPX_MAIN != 0) && \
+                    (defined(__linux) || defined(__linux__) || defined(linux))
+                // make sure the runtime system is not initialized
+                // after its activation from int main()
+                if(get_runtime_ptr() != nullptr &&
```
Comment (hkaiser) [inline src/hpx_init.cpp]: Do we have to check for `get_runtime_ptr() != nullptr` twice?

Comment (NK-Nikunj) [reply src/hpx_init.cpp]: Yes. `include_libhpx_wrap` will only define whether the route of initialization is through main or not. To know whether hpx has already initialized, we will need the help of `get_runtime_ptr()`. I'll merge the 2 if statements to make it more clear.

File: src/hpx_init.cpp [hunk 2]
```diff
@@ -283,6 +283,17 @@ namespace hpx { namespace detail
     }
 }}
 
+#if (HPX_HAVE_DYNAMIC_HPX_MAIN != 0) && \
+    (defined(__linux) || defined(__linux__) || defined(linux))
+namespace hpx_start
+{
+    // Importing weak symbol from libhpx_wrap.a which may be shadowed by one present in
+    // hpx_main.hpp.
+    HPX_SYMBOL_EXPORT __attribute__((weak)) extern bool include_libhpx_wrap;
```
[the remark this answers is not in the document]
Comment (NK-Nikunj) [reply src/hpx_init.cpp]: Yes, we do not need to add attributes twice. I tried to make it clear by adding it twice. I can remove it if you want.

[the remark this answers is not in the document]
Comment (hkaiser) [reply src/hpx_init.cpp]: @NK-Nikunj I'd say either add it everywhere or just for the definition.

**Pull request merged**